File: 06-sorting/01-bubble-sort/01-bubble-sort.rb
def bubble_sort(numbers)
  n = numbers.length
  (0..n-1).each do |i|
    li = 0
    (1..n-i-1).each do |j|
      li = j if numbers[j] > numbers[li]
    end
    numbers[li], numbers[n-i-1] = numbers[n-i-1], numbers[li]
  end
  numbers
end

print bubble_sort([30, 20, 60, 50, 40])

File: 06-sorting/01-bubble-sort/01-bubble-sort.ts
const bubble_sort = (numbers: number[]) => {
  const n = numbers.length
  for (let i = 0; i < n; i++) {
    let [largest_ind, largest_one] = [0, numbers[0]]
    for (let j = 1; j < n - i; j++) {
      if (numbers[j] > largest_one) {
        largest_ind = j
        largest_one = numbers[j]
      }
    }
    const temp = numbers[n - i - 1]
    numbers[n - i - 1] = largest_one
    numbers[largest_ind] = temp
  }
  return numbers
}

console.log(bubble_sort([30, 20, 60, 50, 40]))
console.log(bubble_sort([2, 10, 8, 7]))
console.log(bubble_sort([2]))
